Non-Interactive-Protocol.md

```diff
@@ -0,0 +1,173 @@
+# Non-interactive Onchain Protocol
+
+## Abstract
+
+DLC allows two parties to conduct stateful bitcoin contracts by giving each
+of the parties *cross-signed range of CET transactions*. These CET transactions
+are all spending the same output from a funding transaction and thus only one of
+the alternative can confirm. A CET is finalized according to the outcome signed
+by the previously selected oracle.
+
+As the contract is established once funding transaction signatures have been exchanged,
+from then a party should assume counterparty non-cooperation and ensure either
+confirmation of the funding or double-spend of its collatral inputs to avoid exploitation.
+
+In the optimistic way, once the oracle signature is released, parties interact
+off-chain to settle the DLC. They generate a *closing_tranaction* and publish it on the
+blockchain.
+
+In the pessimitic way, a party interact with the chain to unilaterally settle the DLC.
+They are two scenarios to consider:
+
+1.  The oracle releases the outcome signature but the DLC counterparty goes wrong,
+either going offline or maliciously behaving. The party must broadcast the DLC
+transaction paired with the outcome.
+
+2. The oracle doesn't release the outcome signature, independently of counterparty
+behavior, the party must broadcast the refund transaction after its timelock expiration.
+
+# Table of Contents
+	* [General Requirements](#general-requirements)
+	* [Funding Phase](#funding-phase)
+	* [Execution Phase](#execution-phase)
+	* [Fee-Bumping Policy](#fee-bumping-policy)
+	* [Reorgs] (#reorgs)
+
+# General Requirements
+
+A DLC client should have censorship-resistant access to the blockchain, use a
+local fee-estimator and ensure it's well-connected to the tx-relay network.
+
+We recommend to not react on any mempool event as otherwise it would be a source of
+cross-layer mapping.
+
+We recommend to not sign local transactions until their broadcast is required
+to avoid their misuage in case of site compromise, assuming an external signer.
+
+# Funding Phase
+
+A party always starts the protocol in the funding phase. The funding phase can
+terminate by either a transaction to the execution phase or a protocol abort.
+
+## Requirements
+
+A node:
+	- if a counterparty's funding signature is received:
+		- MUST finalize the funding transaction with a local `SIGHASH_ALL` signature
```

Comment:
Should a node do something if they see their counter party broadcast the tx with a different sig hash type?